test/core/test/jest-expect.test.ts

```diff
@@ -993,6 +977,92 @@ it('toHaveProperty error diff', () => {
   `)
 })
 
+it('toMatchObject error diff', () => {
+  setupColors(getDefaultColors())
+
+  // root key, partial object
+  expect(getError(() => expect({ a: 1, b: 2, c: { d: 4 } }).toMatchObject({ b: 3 }))).toMatchInlineSnapshot(`
+    [
+      "expected { a: 1, b: 2, c: { d: 4 } } to match object { b: 3 }",
+      "- Expected
+    + Received
+
+      Object {
+    -   "b": 3,
+    +   "b": 2,
+      }",
```

Comment:
Can we print something like `... 10 more properties`

Reply:
I assume you mean reporting the number of properties that were stripped from the root of `actual`?

The assertion message printed by Chai already mentions that over a certain threshold. Does it have a lof of added value to mention it again?

For example (`...(2)`).

```ts
expect(getError(() => expect({ a: 1, b: 2, c: { d: 4 }, x: 5, y: 6, z: 7 }).toMatchObject({ a: 1, b: 2, c: { d: 5 } }))).toMatchInlineSnapshot(`
    [
      "expected { a: 1, b: 2, c: { d: 4 }, x: 5, …(2) } to match object { a: 1, b: 2, c: { d: 5 } }",
      "- Expected
    + Received

      Object {
        "a": 1,
        "b": 2,
        "c": Object {
    -     "d": 5,
    +     "d": 4,
        },
      }",
    ]
  `)
```

Reply:
Yes, but it won't show it in nested objects, would it?

Reply:
I tweaked the message to list the number of properties that were stripped from the actual (incl. all the nested) ones. There are some examples in the `jest-expect.ts.ts`.